src/Model/File.php

```diff
@@ -122,6 +167,125 @@ public function createFromPath(string $path, string $fileName = null): Model
 
         $entity->save();
 
+        // create thumbnail images if needed and possible
+        if ($entity->get('meta_is_image') && $this->createThumbnail === true) {
+            $entity->createThumbnail($path);
+        }
+
         return $entity;
     }
+
+    /**
+     * Create thumbnail images of current image.
+     *
+     * @param string $path Local path to original file
+     *
+     * @return bool True on success
+     */
+    public function createThumbnail(string $path): bool
+    {
+        $this->assertIsEntity();
+
+        $max_width = $this->thumbnailMaxWidth;
+        $max_height = $this->thumbnailMaxHeight;
+
+        [$width, $height, $image_type] = getimagesize($path);
+
+        switch ($image_type) {
```

Comment:
can `imagecreatefromstring` be used here?

Reply:
yes it can, addressed